nexus/src/sagas.rs

```diff
@@ -223,3 +246,146 @@ async fn sic_instance_ensure(
         .map(|_| ())
         .map_err(ActionError::action_failed)
 }
+
+/*
+ * "Migrate Instance" saga template
+ */
+#[derive(Debug, Deserialize, Serialize)]
+pub struct ParamsInstanceMigrate {
+    // TODO design: We can't just use db::model::Instance here as it doesn't impl
+    // Serialize so instead we need the project_id/instance_name to look it up
+    // in the saga.
+    pub project_id: Uuid,
+    pub instance_name: Name,
+    pub migrate_params: params::InstanceMigrate,
+}
+
+#[derive(Debug)]
+pub struct SagaInstanceMigrate;
+impl SagaType for SagaInstanceMigrate {
+    type SagaParamsType = Arc<ParamsInstanceMigrate>;
+    type ExecContextType = Arc<SagaContext>;
+}
+
+#[derive(Copy, Clone, Debug, Deserialize, Serialize)]
+pub enum MigrateType {
+    /// Migrate to a new propolis instance on a different sled
+    Sled,
+    /// Migrate to a new propolis instance on the same sled
+    InPlace,
+}
```

Comment:
Are we actually using this enum anymore / do we have a need for the two distinct templates? Kinda seems like we could get away with just the one, if I'm reading this source correctly.

Reply:
Yea, I was mulling about keeping it if needed later but can always easily add it back.